tests/test_config.py

```
@@ -105,6 +105,43 @@ def test_api_key_env_unknown_provider_raises(tmp_path):
         config.load_from_repo(tmp_path)
 
 
+# ── Internal validator edge cases ─────────────────────────────────────────────
+
+def test_require_type_raises_config_error(tmp_path):
+    # tasks.check must be a dict; passing a string triggers _require_type
+    _write(tmp_path / "forerunner.config.yaml", "tasks:\n  check: not-a-dict\n")
+    with pytest.raises(config.ConfigError, match="tasks.check"):
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)